src/appleseed/renderer/modeling/input/colorsource.cpp

```diff
@@ -53,14 +53,14 @@ namespace renderer
 // ColorSource class implementation.
 //
 
-ColorSource::ColorSource(const ColorEntity& color_entity)
+ColorSource::ColorSource(const ColorEntity& color_entity, const char* intent)
   : Source(true)
   , m_color_entity(color_entity)
 {
     // Retrieve the color values.
     if (color_entity.get_color_space() == ColorSpaceSpectral)
-        initialize_from_spectrum(color_entity);
-    else initialize_from_color3(color_entity);
+        initialize_from_spectrum(color_entity, std::strcmp(intent, "reflectance") == 0 ? Spectrum::Reflectance : Spectrum::Illuminance);
```

Comment:
I don't like it. I will try to do some refactoring and remove this string comparison.